src/elfcore.cc

```diff
@@ -290,6 +296,18 @@ class SysCalls {
   do {                 \
   } while ((fn) < 0 && ERRNO == EINTR)
 
+/* Since we know the proc path for self is constant we don't want to
```

Comment:
I'm not convinced this is a big deal compared to dumping core of the process, and it means you have to pass around a bunch of extra args, rather than a single "ext_pid" value that can be 0 for the self case. Please consider switching to a normal function and using similar code for both cases (though advising strcat).

Reply:
Introduced the function that append the file name to proc path and open the file, then return the fd.